app/javascript/react/components/labware_content_input.jsx

```diff
@@ -0,0 +1,156 @@
+import React from "react"
+import { connect } from 'react-redux'
+import StateSelectors from '../selectors'
+import { setManifestValue, saveTab, updateScientificName } from '../actions'
+import { debounce, throttle } from 'throttle-debounce'
+
+const logName = (name) => {  }
+
+const DEBOUNCED_TIMING=500
+
+export const LabwareContentSelectComponent = (props) => {
+  logName('LabwareContentSelectComponent')
+  const {fieldName,title,name,id,selectedOptionValue,onChange} = props
+
+  return(
+    <select onChange={onChange} className="form-control" title={title} name={name} id={id} value={selectedOptionValue}>
+      <option value=""></option>
+      { props.optionsForSelect.map((val, pos) => {
+        return (<option key={pos} value={val}>{val}</option>)
+      }) }
+    </select>
+  )
+}
+
+export const LabwareContentSelect = connect((state, ownProps) => {
+  return {
+    selectedOptionValue: StateSelectors.schema.selectedOptionValue(state, ownProps.fieldName, ownProps.selectedValue),
+    optionsForSelect: StateSelectors.schema.optionsForSelect(state, ownProps.fieldName)
+  }
+})(LabwareContentSelectComponent)
+
+export const LabwareContentText = (props) => {
+  logName('LabwareContentText')
+  const {onChange,selectedValue,title,name,id} = props
+  return(
+    <input onChange={onChange}
+      className="form-control" title={title} name={name} id={id}
+      value={selectedValue} />
+  )
+}
+
+class LabwareContentInputComponent extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      value: props.selectedValue,
+      stateValueSelected: 'redux'
+    }
+    this.onChangeProcessThrottelledCall = this.buildThrottelledCall()
+  }
+
+  setStateValueSelected(str) {
+    console.log('set value to '+str)
```

Comment:
The logging in this page should be wrapped in a debug/devel environment checker or removed.